src/pkg/backup/details/testdata/testdata.go

```diff
@@ -33,7 +33,97 @@ func mustAppendPath(p path.Path, newElement string, isItem bool) path.Path {
 	return newP
 }
 
+func locFromRepo(rr path.Path, isItem bool) *path.Builder {
+	loc := &path.Builder{}
+
+	for _, e := range rr.Folders() {
+		loc = loc.Append(strings.TrimSuffix(e, folderSuffix))
+	}
+
+	if rr.Service() == path.OneDriveService || rr.Category() == path.LibrariesCategory {
+		loc = loc.PopFront()
+	}
+
+	// Folders don't have their final element in the location.
+	if !isItem {
+		loc = loc.Dir()
+	}
+
+	return loc
+}
+
+type repoRefAndLocRef struct {
+	RR  path.Path
+	loc *path.Builder
+}
+
+func (p repoRefAndLocRef) mustAppend(newElement string, isItem bool) repoRefAndLocRef {
+	e := newElement + folderSuffix
+
+	if isItem {
+		e = newElement + fileSuffix
+	}
+
+	res := repoRefAndLocRef{
+		RR: mustAppendPath(p.RR, e, isItem),
+	}
+
+	res.loc = locFromRepo(res.RR, isItem)
+
+	return res
+}
+
+func (p repoRefAndLocRef) ItemLocation() string {
+	return strings.TrimSuffix(p.RR.Item(), fileSuffix)
+}
+
+func (p repoRefAndLocRef) FolderLocation() string {
+	lastElem := p.RR.ToBuilder().LastElem()
+
+	if len(p.RR.Item()) > 0 {
+		f := p.RR.Folders()
+		lastElem = f[len(f)-2]
+	}
+
+	return p.loc.Append(strings.TrimSuffix(lastElem, folderSuffix)).String()
+}
+
+func mustPathRep(ref string, isItem bool) repoRefAndLocRef {
+	res := repoRefAndLocRef{}
+	tmp := mustParsePath(ref, isItem)
+
+	// Now append stuff to the RepoRef elements so we have distinct LocationRef
+	// and RepoRef elements to simulate using IDs in the path instead of display
+	// names.
+	rrPB := &path.Builder{}
+	for _, e := range tmp.Folders() {
+		rrPB = rrPB.Append(e + folderSuffix)
+	}
+
+	if isItem {
+		rrPB = rrPB.Append(tmp.Item() + fileSuffix)
+	}
+
+	rr, err := rrPB.ToDataLayerPath(
+		tmp.Tenant(),
+		tmp.ResourceOwner(),
+		tmp.Service(),
+		tmp.Category(),
+		isItem)
+	if err != nil {
+		panic(err)
+	}
+
+	res.RR = rr
+	res.loc = locFromRepo(rr, isItem)
+
+	return res
+}
+
 const (
+	folderSuffix = ".d"
+	fileSuffix   = ".f"
```

Comment:
seeing this in both details and in selectors kinda makes me want a shared owner.  tester?  That's probably too heavy handed of me.

Reply:
I think long-term we should [re]move a good chunk of tests for selectors. TBH I'm not interested in testing every function. I'm interested in the system working for the given inputs and outputs (black box testing). Testing every function in the selectors package provides minimal benefits as it ends up testing large chunks of the code multiple times while generating lots of work to make even minor changes (as seen in this PR)

Moving tests that check edge cases to the test data in the details package would go a long way to reducing the overhead for changes while also reducing duplicate work in tests

My general rule of thumb is not to test unexported functions if
* tests on exported functions allow good coverage of unexported code paths
* not having tests for the unexported functions will not greatly hinder debugging of other test failures (i.e. unexported functions aren't too complex)

If other packages start wanting to use data like this we should centralize the data definitions, not the bits that people may use to build their own version of the data. That will help cut down on mismatches in the expectations of different modules when stuff like the separation between RepoRef and LocationRef goes in

Reply:
> TBH I'm not interested in testing every function. I'm interested in the system working for the given inputs and outputs (black box testing). 

I'm not on board with 100% test coverage, either.  But the more fiddly-bits there are to mess up, the more need there is to ensure implementation gets all the fiddly bits right.  And, as we both know, selectors has many fiddly bits.  So the complaint is really about the core design, and the symptom involves the test requirements surfaced by that design.  I'm sure you agree about that.  Still, per `we should [re]move a good chunk of tests for selectors`, before we attack the symptom, I think we should first investigate a design that removes the underlying _need_ to have this granularity of testing.